cli/hooks/webpack.js

```diff
@@ -0,0 +1,129 @@
+// eslint-disable-next-line security/detect-child-process
+const { execSync } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const semver = require('semver');
+const util = require('util');
+require('colors');
+
+const WebpackService = require('../lib/webpack/service');
+
+const MIN_APPCD_VERSION = '3.1.0';
+let AppcdClient;
+
+exports.id = 'ti.webpack';
+exports.init = (logger, config, cli) => {
+	let isBuildCommand = false;
+	let isWebpackProject = false;
+
+	cli.on('cli:command-loaded', (hookData) => {
+		const command = hookData.command;
+		if (command.name !== 'build') {
+			return;
+		}
+
+		isBuildCommand = true;
+	});
+
+	cli.on('cli:post-validate', () => {
+		if (!isBuildCommand) {
+			return;
+		}
+
+		isWebpackProject = !!cli.tiapp.webpack;
+		if (!isWebpackProject) {
+			return;
+		}
+
+		let appcdRootPath;
+		if (process.env.APPC_ENV) {
+			// we were invoked by appc-cli, load bundled daemon client.
+			appcdRootPath = resolveModuleRoot('appcd');
+			if (!fs.existsSync(appcdRootPath)) {
+				throw new Error('Unable to find Appcelerator Daemon inside the appc-cli. Please make sure to use a recent appc-cli version. You can install/select it with  "appc use latest".');
+			}
+			ensureAppcdVersion(appcdRootPath, MIN_APPCD_VERSION, true);
+		} else {
+			// plain ti-cli, load daemon from global modules
+			const globalModulesPath = execSync('npm root -g').toString().replace(/\s+$/, '');
+			appcdRootPath = resolveModuleRoot('appcd', [ globalModulesPath ]);
+			if (!fs.existsSync(appcdRootPath)) {
+				throw new Error('Unable to find global Appcelerator Daemon install. You can install it with "npm i appcd -g".');
+			}
+			ensureAppcdVersion(appcdRootPath, MIN_APPCD_VERSION);
+		}
+
+		AppcdClient = resolveAppcdClient(appcdRootPath);
+	});
+
+	cli.on('build.pre.compile', {
+		priority: 800,
+		post(builder, callback) {
+			if (!isWebpackProject) {
+				return callback();
+			}
+
+			builder.useWebpack = true;
+
+			const webpackLogger = {
+				info(...args) {
+					logger.info(this.formatWithBadge(args));
+				},
+				debug(...args) {
+					logger.debug(this.formatWithBadge(args));
+				},
+				formatWithBadge(args) {
+					return `${' WEBPACK '.bgCyan.black} ${util.format(...args)}`;
+				}
+			};
+
+			const webpackService = new WebpackService({
+				AppcdClient,
+				logger: webpackLogger,
+				cli,
+				builder
+			});
+			Promise.resolve().then(async () => {
+				await webpackService.ensureDaemonIsRunning();
+				await webpackService.ensureWebpackIsRunning();
+				await webpackService.waitForCompilationComplete();
+				return callback();
```

Comment:
<!--
  0 failure: 
  1 warning:  cli/hooks/webpack...
  
  
  DangerID: danger-id-Danger;
  File: cli/hooks/webpack.js;
  Line: 90;
-->

- :warning: cli/hooks/webpack.js line 90 – Avoid calling back inside of a promise. (promise/no-callback-in-promise)